packages/jss/src/RuleList.js

```diff
@@ -180,7 +180,10 @@ export default class RuleList {
   /**
    * Execute plugins, update rule props.
    */
-  onUpdate(data: Object, rule: Rule, options?: Object = defaultUpdateOptions) {
+  onUpdate(data: Object, rule?: Rule, options?: Object = defaultUpdateOptions) {
+    // Should be undefined during hot-reloading
+    if (!rule) return
```

Comment:
I guess this makes sense as we are typing it as optional anyway. Not sure how flow would even allow something like this

Reply:
Flow will have a problem here, rule isn't optional. We need a clear reproduction of this case to understand how rule can be undefined and we need to fix it far before, so that it doesn't come to this state.